### PR TITLE
Add retry for sending telemetry to Helix

### LINKED ISSUE #1781: Arcade Helix telemetry sending code needs to retry on HTTP connection error
These builds had no actual problems, just some sort of network burp happened sending telemetry:

https://dev.azure.com/dnceng/internal/_build/results?buildId=71442
https://dev.azure.com/dnceng/internal/_build/results?buildId=71437

I'm not 100% I support failing the build in this scenario, but at a minimum we should retry for a minute or so before giving up; helix telemetry for job start and end is safe to duplicate already.

### PR DESCRIPTION
**Pull request by jonfortescue**
Fixes #1781 by adding some basic retry to the telemetry sending scripts.

### CODE REVIEW

File: eng/common/templates/steps/telemetry-end.yml [hunk 1]
```diff
@@ -7,24 +7,38 @@ steps:
...
-    curlStatus=$?
-
-    if [ $curlStatus -eq 0 ]; then
-      if [ $curlResult -gt 299 ] || [ $curlResult -lt 200 ]; then
-        curlStatus=$curlResult
+    curlStatus=1
+    retryCount=0
+    # retry loop to harden against spotty telemetry connections
+    # we don't retry successes and 4xx client errors
+    until [[ $curlStatus -eq 0 || ( $curlStatus -ge 400 && $curlStatus -le 499 ) || $retryCount -gt 5 ]]
```
Comment (MattGal) [inline eng/common/templates/steps/telemetry-end.yml]: probably useful to stick that 5 into a variable like "maxRetries" ?

File: eng/common/templates/steps/telemetry-end.yml [hunk 2]
```diff
@@ -7,24 +7,38 @@ steps:
...
       fi
-    fi
+
+      # create a temporary file for curl output
+      res=`mktemp`
+
+      curlResult=`
+        curl --verbose --output $res --write-out "%{http_code}"\
+        -H 'Content-Type: application/json' \
+        -H "X-Helix-Job-Token: $Helix_JobToken" \
```
Comment (MattGal) [inline eng/common/templates/steps/telemetry-end.yml]: Curious... is it intentional we mix pascal and camel casing?

File: eng/common/templates/steps/telemetry-end.yml (hunk 1 above)
Comment (jonfortescue) [reply eng/common/templates/steps/telemetry-end.yml]: yeah probably for the best.

File: eng/common/templates/steps/telemetry-end.yml (hunk 2 above)
Comment (jonfortescue) [reply eng/common/templates/steps/telemetry-end.yml]: I don't know. @chcosta would be the best person to ask; for now I'm just matching the standards already here.

Comment (chcosta) [reply eng/common/templates/steps/telemetry-end.yml]: Not to blame-shift, but... @alexperovich originally wrote this.  That being said, this is utilizing variables from a variety of locations (Azure DevOps variable groups, Azure DevOps environment variables, and defined powershell variables, etc...).

Arguably, there could be some cleanup (ie, is "res" a good variable name?).

All of this should be replaced by an Azure DevOps task relatively soon and I don't think it's worth fixing at this time.

Comment (jonfortescue) [reply eng/common/templates/steps/telemetry-end.yml]: Yeah I'm going to say out-of-scope for now anyway and resolve this.

**Pull request merged**